client/src/reducers/page_location.coffee

```diff
@@ -0,0 +1,18 @@
+{ GO_TO_LOG_IN, LOG_IN, LOG_OUT, GO_TO_HOME, GO_TO_HELP } = require '../constants/action_types.coffee'
+
+Immutable = require 'immutable'
+
+initialState = Immutable.fromJS
+  currentPage: 'home'
+
+pageLocation = (state = initialState, action) ->
+  switch action.type
+    when LOG_IN, LOG_OUT, GO_TO_HOME
+      state.merge currentPage: 'home'
```

Comment:
I actually don't think extracting these to their own functions at this stage makes the code clearer. Most of the time an additional abstraction makes the behavior much clearer, but these functions are destined to be named things like "onLogIn" or "onLogOut", so the abstraction won't actually help.

When these behaviors become more sophisticated I think i could support a full extraction for each of these functions, but I can't support it for more than the login action at the moment.